### PR TITLE
feat: login and signup

### PR DESCRIPTION
**Pull request by rohanmathur91**
## Description:
- Added login and logout feature
- Added option to login with test credentials
- Added signup feature
- Did refactor to handle login and signup form using custom hook
- Added basic profile page

## Link:
https://deploy-preview-9--evolt-prime.netlify.app/

## Screenshot:
chrome-capture (9)

### CODE REVIEW

File: src/hooks/useAuthForm.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useReducer } from "react";
+import { authFormReducer } from "../reducers";
+
+export const useAuthForm = (initialForm) => {
+  const [authFormState, authFormDispatch] = useReducer(authFormReducer, {
+    loading: false,
+    showPassword: false,
+    credentials: initialForm,
```
Comment (anupkgurung) [inline src/hooks/useAuthForm.js:8]: Suggestion assuming initialForm is related to user credentials you could use some more specific name maybe something sort of initial values or default.

**Review by anupkgurung — APPROVED** [review]
good use of context reducer and custom hooks.
clean and thoughtful code structure.
Note: not sure if importing css to app.css is correct.

File: src/hooks/useAuthForm.js (hunk 1 above)
Comment (rohanmathur91) [reply src/hooks/useAuthForm.js:8]: l will update that

File: src/pages/Login/Login.jsx [hunk 2]
```diff
@@ -1,5 +1,124 @@
...
+    email: "",
+    password: "",
+  });
+
+  const navigate = useNavigate();
+  const { updateUser } = useAuth();
+  const { playlistDispatch } = usePlaylist();
+
+  useScrollToTop();
+  useDocumentTitle("Login");
```
Comment (hrishikeshmane) [inline src/pages/Login/Login.jsx:27]: These are side effects. Move them to useEffect

File: src/pages/Signup/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    password: "",
+    confirmPassword: "",
+  });
+  const [errorState, errorDispatch] = useReducer(
+    signupErrorReducer,
+    signUpErrorInitialState
+  );
+
+  useScrollToTop();
+  useDocumentTitle("Signup");
```
Comment (hrishikeshmane) [inline src/pages/Signup/Signup.jsx:33]: Same here.

**Review by hrishikeshmane — COMMENTED** [review]
Ideally local storage is also considered as side effects and should be handled in useEffect. But you can do this in your next sprint. 

Seems ok for now.

File: src/pages/Login/Login.jsx (hunk 2 above)
Comment (rohanmathur91) [reply src/pages/Login/Login.jsx:27]: those are custom hooks, which are using useEffect only

**Review by hrishikeshmane — APPROVED** [review]

**Pull request merged**